### PR TITLE
Web app now supports SSO config

### PR DESCRIPTION
**Pull request by douxc**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

you can set enable/disable sso feature for web app at setting modal

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

### CODE REVIEW

File: web/i18n/zh-Hans/app-overview.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ const translation = {
         chatColorThemeDesc: '设置聊天机器人的颜色主题',
         chatColorThemeInverted: '反转',
         invalidHexMessage: '无效的十六进制值',
+        sso: {
+          label: '认证设置',
```
Comment (GarfieldDai) [inline web/i18n/zh-Hans/app-overview.ts]: 单点登录认证

File: web/i18n/en-US/app-overview.ts [hunk 2]
```diff
@@ -53,6 +53,12 @@ const translation = {
         chatColorThemeDesc: 'Set the color theme of the chatbot',
         chatColorThemeInverted: 'Inverted',
         invalidHexMessage: 'Invalid hex value',
+        sso: {
+          label: 'SSO ENFORCEMENT',
+          title: 'Web Application SSO',
```
Comment (GarfieldDai) [inline web/i18n/en-US/app-overview.ts]: WebApp SSO

File: web/i18n/zh-Hans/app-overview.ts [hunk 3]
```diff
@@ -53,6 +53,12 @@ const translation = {
...
+          title: 'WebApp 认证',
```
Comment (GarfieldDai) [inline web/i18n/zh-Hans/app-overview.ts]: WebApp SSO 认证

File: web/i18n/zh-Hans/app-overview.ts [hunk 4]
```diff
@@ -53,6 +53,12 @@ const translation = {
...
+          description: '启用后，所有用户都需要先认证后才能访问',
+          tooltip: '联系管理员以开启 Web 应用的 SSO 认证',
```
Comment (GarfieldDai) [inline web/i18n/zh-Hans/app-overview.ts]: 联系管理员以开启 WebApp SSO 认证

File: web/i18n/en-US/app-overview.ts [hunk 5]
```diff
@@ -53,6 +53,12 @@ const translation = {
...
+          description: 'All users are required to login with SSO before using web application',
+          tooltip: 'Contact the administrator to enable Web Application SSO',
```
Comment (GarfieldDai) [inline web/i18n/en-US/app-overview.ts:60]: Contact the administrator to enable WebApp SSO

File: web/i18n/en-US/app-overview.ts [hunk 6]
```diff
@@ -53,6 +53,12 @@ const translation = {
         chatColorThemeDesc: 'Set the color theme of the chatbot',
         chatColorThemeInverted: 'Inverted',
         invalidHexMessage: 'Invalid hex value',
+        sso: {
+          label: 'SSO ENFORCEMENT',
+          title: 'Web Application SSO',
+          description: 'All users are required to login with SSO before using web application',
```
Comment (GarfieldDai) [inline web/i18n/en-US/app-overview.ts]: All users are required to login with SSO before using WebApp

File: web/i18n/en-US/app-overview.ts (hunk 5 above)
Comment (GarfieldDai) [inline web/i18n/en-US/app-overview.ts]: Contact the administrator to enable WebApp SSO

File: web/i18n/zh-Hans/app-overview.ts [hunk 7]
```diff
@@ -53,6 +53,12 @@ const translation = {
         chatColorThemeDesc: '设置聊天机器人的颜色主题',
         chatColorThemeInverted: '反转',
         invalidHexMessage: '无效的十六进制值',
+        sso: {
+          label: '认证设置',
+          title: 'WebApp 认证',
+          description: '启用后，所有用户都需要先认证后才能访问',
```
Comment (GarfieldDai) [inline web/i18n/zh-Hans/app-overview.ts]: 启用后，所有用户都需要先进行 SSO 认证才能访问

File: web/app/(commonLayout)/app/(appDetailLayout)/[appId]/overview/cardView.tsx [hunk 8]
```diff
@@ -28,10 +31,18 @@ const CardView: FC<ICardViewProps> = ({ appId }) => {
   const { notify } = useContext(ToastContext)
   const appDetail = useAppStore(state => state.appDetail)
   const setAppDetail = useAppStore(state => state.setAppDetail)
+  const systemFeatures = useContextSelector(AppContext, state => state.systemFeatures)
 
   const updateAppDetail = async () => {
     fetchAppDetail({ url: '/apps', id: appId }).then((res) => {
-      setAppDetail(res)
+      if (systemFeatures.enable_web_sso_switch_component) {
```
Comment (crazywoola) [inline web/app/(commonLayout)/app/(appDetailLayout)/[appId]/overview/cardView.tsx:39]: Avoid using nested promises in async function, you should use `try catch` to handle the exceptions as well.

File: web/app/components/app/overview/settings/index.tsx [hunk 9]
```diff
@@ -138,6 +146,7 @@ const SettingsModal: FC<ISettingsModalProps> = ({
       icon: emoji.icon,
       icon_background: emoji.icon_background,
       show_workflow_steps: inputInfo.show_workflow_steps,
+      enable_sso: inputInfo.enable_sso!,
```
Comment (crazywoola) [inline web/app/components/app/overview/settings/index.tsx]:
```
{
    title,
    desc: description,
    chatColorTheme: chat_color_theme,
    chatColorThemeInverted: chat_color_theme_inverted,
    copyright,
    privacyPolicy: privacy_policy,
    customDisclaimer: custom_disclaimer,
    show_workflow_steps,
    enable_sso: appInfo.enable_sso,
  }
```
This should have a type, and make `enable_sso` optional.

File: web/app/components/base/tooltip-plus/index.tsx [hunk 10]
```diff
@@ -67,7 +69,7 @@ const Tooltip: FC<TooltipProps> = ({
 
   return (
     <PortalToFollowElem
-      open={open}
+      open={disabled ? false : open}
```
Comment (crazywoola) [inline web/app/components/base/tooltip-plus/index.tsx]:
```suggestion
      open={!disabled && open}
```

File: web/i18n/en-US/app-overview.ts [hunk 11]
```diff
@@ -53,6 +53,12 @@ const translation = {
         chatColorThemeDesc: 'Set the color theme of the chatbot',
         chatColorThemeInverted: 'Inverted',
         invalidHexMessage: 'Invalid hex value',
+        sso: {
+          label: 'SSO ENFORCEMENT',
```
Comment (crazywoola) [inline web/i18n/en-US/app-overview.ts]: This is a wrong for `'单点登录认证'`

File: web/service/apps.ts [hunk 12]
```diff
@@ -13,6 +13,13 @@ export const fetchAppDetail = ({ url, id }: { url: string; id: string }) => {
   return get<AppDetailResponse>(`${url}/${id}`)
 }
 
+export const fetchAppSSO = async ({ appId }: { appId: string }) => {
+  return get<AppSSOResponse>(`/enterprise/app-setting/sso?appID=${appId}`)
+}
+export const updateAppSSO = async ({ id, enabled }: { id: string;enabled: boolean }) => {
```
Comment (crazywoola) [inline web/service/apps.ts]:
```suggestion
export const updateAppSSO = async ({ id, enabled }: { id: string; enabled: boolean }) => {
```

**Review by crazywoola — CHANGES_REQUESTED** [review]
See comments

**Review by crazywoola — APPROVED** [review]
LGTM cc @GarfieldDai

**Review by GarfieldDai — APPROVED** [review]
LGTM

**Pull request merged**